File: src/main/java/com/enigmacamp/testwebmvc/entity/Employee.java
package com.enigmacamp.testwebmvc.entity;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "t2_employee")
    public class Employee {

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        private Integer id;

        private String name;

        @Column(name = "birth_date")
        private LocalDate birthDate;

        @ManyToOne(fetch = FetchType.EAGER)
        @JoinColumn(name = "position_id")
        private Position position;

        @Column(name = "id_number")
        private Integer idNumber;

        private Integer gender;

        @Column(name = "is_delete", nullable = false)
        private Integer isDelete = 0;
}

File: test-web-frontend/src/types/Employee.ts
export type Gender = "MALE" | "FEMALE";

export interface Employee {
  id: number;
  name: string;
  birthDate: string;
  idNumber: number;
  gender: Gender;
  positionId: number;
  positionName: string;
}

export interface EmployeeRequest {
  name: string;
  birthDate: string;
  idNumber: number;
  gender: Gender;
  positionId: number;
}